android/app/src/main/java/com/microsoft/reacttestapp/Session.kt

```diff
@@ -0,0 +1,44 @@
+package com.microsoft.reacttestapp
+
+import android.content.Context
+import android.content.Context.MODE_PRIVATE
+import android.content.SharedPreferences
+import androidx.core.content.edit
+
+class Session(context: Context) {
+
+    private companion object {
```

Comment:
You don't need to use companion objects for private constants. You can simply extract them from the class to the file level and make them private.

Reply:
Getting this:
```
e: /~/android/app/src/main/java/com/microsoft/reacttestapp/Session.kt: (10, 13): Const 'val' are only allowed on top level or in objects
e: /~/android/app/src/main/java/com/microsoft/reacttestapp/Session.kt: (11, 13): Const 'val' are only allowed on top level or in objects
e: /~/android/app/src/main/java/com/microsoft/reacttestapp/Session.kt: (12, 13): Const 'val' are only allowed on top level or in objects
```
Are we fine with not making them consts?

Reply:
You need to extract them to the file level:

```kotlin
private const val MANIFEST_CHECKSUM = "ManifestChecksum"
private const val REMEMBER_LAST_COMPONENT_ENABLED = "RememberLastComponent/Enabled"
private const val REMEMBER_LAST_COMPONENT_INDEX = "RememberLastComponent/Index"

class Session(context: Context) {
  ...
}
```

Reply:
Nice 👍